Source/Core/DolphinNoGUI/PlatformMacos.mm

```diff
@@ -0,0 +1,426 @@
+  [m_window setDelegate:m_window_delegate];
+
+  NSNotificationCenter* c = [NSNotificationCenter defaultCenter];
+  [c addObserver:NSApp
+        selector:@selector(shutdown)
+            name:NSWindowWillCloseNotification
+          object:m_window];
+
+  if (m_window == nil)
+  {
+    NSLog(@"Window is %@\n", m_window);
+    return false;
+  }
+
+  if (Config::Get(Config::MAIN_SHOW_CURSOR) == Config::ShowCursor::Never)
+    [NSCursor hide];
+
+  if (Config::Get(Config::MAIN_FULLSCREEN))
+  {
+    m_window_fullscreen = true;
+    [m_window toggleFullScreen:m_window];
+  }
+
+  [m_window makeKeyAndOrderFront:NSApp];
+  [m_window makeMainWindow];
+  [NSApp activateIgnoringOtherApps:YES];
+  [m_window setTitle:@"Dolphin-emu-nogui"];
+
+  SetupMenu();
+
+  return true;
+}
+
+void PlatformMacOS::SetTitle(const std::string& title)
+{
+  @autoreleasepool
+  {
+    NSWindow* window = m_window;
+    NSString* str = [NSString stringWithUTF8String:title.c_str()];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [window setTitle:str];
+    });
+  }
+}
+
+void PlatformMacOS::MainLoop()
+{
+  while (IsRunning())
+  {
+    UpdateRunningFlag();
+    Core::HostDispatchJobs();
+    ProcessEvents();
+    UpdateWindowPosition();
+  }
+}
+
+WindowSystemInfo PlatformMacOS::GetWindowSystemInfo() const
+{
+  @autoreleasepool
+  {
+    WindowSystemInfo wsi;
+    wsi.type = WindowSystemType::MacOS;
+    wsi.render_window = (void*)CFBridgingRetain([m_window contentView]);
+    wsi.render_surface = wsi.render_window;
+    return wsi;
+  }
+}
+
+void PlatformMacOS::ProcessEvents()
+{
+  @autoreleasepool
+  {
+    NSDate* expiration = [NSDate dateWithTimeIntervalSinceNow:1];
+    NSEvent* event = [NSApp nextEventMatchingMask:NSEventMaskAny
+                                        untilDate:expiration
+                                           inMode:NSDefaultRunLoopMode
+                                          dequeue:YES];
+
+    [NSApp sendEvent:event];
+
+    // Need to update if m_window becomes fullscreen
+    m_window_fullscreen = [m_window styleMask] & NSWindowStyleMaskFullScreen;
+
+    if ([m_window isMainWindow])
+    {
+      m_window_focus = true;
+      if (Config::Get(Config::MAIN_SHOW_CURSOR) == Config::ShowCursor::Never &&
+          Core::GetState() != Core::State::Paused)
+        [NSCursor unhide];
+    }
+    else
+    {
+      m_window_focus = false;
+      if (Config::Get(Config::MAIN_SHOW_CURSOR) == Config::ShowCursor::Never)
+        [NSCursor hide];
+    }
+  }
+}
+
+void PlatformMacOS::UpdateWindowPosition()
+{
+  if (m_window_fullscreen)
+    return;
+
+  NSRect win = [m_window frame];
+  m_window_x = win.origin.x;
+  m_window_y = win.origin.y;
+  m_window_width = win.size.width;
+  m_window_height = win.size.height;
+}
+
+void PlatformMacOS::SetupMenu()
+{
+  @autoreleasepool
+  {
+    menuBar = [NSMenu new];
+
+    NSMenu* appMenu = [NSMenu new];
+    NSMenu* stateMenu = [[NSMenu new] initWithTitle:@"States"];
```

Comment:
`[xxx new]` is the equivalent of `[[xxx alloc] init]`, so you want `[[NSMenu alloc] initWithTitle:]`for these, otherwise you're doing the equivalent of `[[[NSMenu alloc] init] initWithTitle:]`

Reply:
I wouldn't need it on the first two since I'm not giving them a title, but yeah for stateMenu I should fix that